cpp/src/arrow/adapters/orc/adapter_test.cc

```diff
@@ -15,20 +15,39 @@
 // specific language governing permissions and limitations
 // under the License.
 
-#include <string>
-
 #include "arrow/adapters/orc/adapter.h"
-#include "arrow/array.h"
-#include "arrow/io/api.h"
 
 #include <gtest/gtest.h>
+
 #include <orc/OrcFile.hh>
+#include <string>
+
+#include "arrow/adapters/orc/adapter_util.h"
+#include "arrow/api.h"
+#include "arrow/array.h"
+#include "arrow/io/api.h"
+#include "arrow/io/interfaces.h"
+#include "arrow/io/memory.h"
+#include "arrow/testing/gtest_util.h"
+#include "arrow/testing/random.h"
+#include "arrow/type.h"
+#include "arrow/util/decimal.h"
 
 namespace liborc = orc;
 
 namespace arrow {
 
-constexpr int DEFAULT_MEM_STREAM_SIZE = 100 * 1024 * 1024;
+constexpr int kDefaultSmallMemStreamSize = 16384 * 5;  // 80KB
+constexpr int kDefaultMemStreamSize = 10 * 1024 * 1024;
+static constexpr random::SeedType kRandomSeed = 0x0ff1ce;
+
+using ArrayBuilderVector = std::vector<std::shared_ptr<ArrayBuilder>>;
```

Comment:
I think these aliases would probably be less common if the comments below about reduce test size are used.  My preference would be to not use them (we use such aliases sparingly in other parts of the code.)